### PR TITLE
Validate Project Names

### PR DESCRIPTION
**Pull request by milesthedisch**
**What kind of change does this PR introduce?**

Add test and validation to `dest` and/or `name` options.

**Did you add tests for your changes?**

Yes

**Summary**

https://github.com/developit/preact-cli/issues/466

**Does this PR introduce a breaking change?**

No

**Other information**

- node v6.10.3
- npm 3.10.10
- yarn 1.3.2
- macOS Sierra 10.12.6

### CODE REVIEW

File: src/commands/create.js [hunk 1]
```diff
@@ -89,6 +90,16 @@ export default asyncCommand({
...
 		}
 
+		// Use `--name` value or `dest` dir's name
+		argv.name = argv.name || argv.dest;
+
+		const npmPackageNameErrors = validateNpmPackageName(argv.name).errors;
+
+		if (npmPackageNameErrors) {
+			const npmPackageNameMsg = npmPackageNameErrors.join(', ');
+			return error(`${argv.name} is not a valid package name. ${npmPackageNameMsg}`, 1);
```
Comment (prateekbh) [inline src/commands/create.js]: can you add a `chalk` message before this?

Comment (milesthedisch) [reply src/commands/create.js]: Not sure what the chalk message would have in it. Doesn't the `error` method from the utils already display the message given?

Comment (reznord) [reply src/commands/create.js]: You are right. Internally `error` method in `utils` uses chalk itself. 

@prateekbh that is totally valid. 😄

**Comment by lukeed** [comment]
Thank you @milesthedisch ~!

File: src/commands/create.js (hunk 1 above)
Comment (prateekbh) [reply src/commands/create.js]: gosh I need to do my revision for this codebase

**Pull request merged**